src/test/socket/getsockname/test_getsockname.rs

```diff
@@ -0,0 +1,713 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+struct GetsocknameArguments {
+    fd: libc::c_int,
+    addr: Option<libc::sockaddr_in>, // if None, a null pointer should be used
+    addr_len: Option<libc::socklen_t>, // if None, a null pointer should be used
+}
+
+/// A boxed function to run as a test.
+type TestFn = Box<dyn Fn() -> Result<(), String>>;
+
+trait AsPtr<T> {
+    fn as_ptr(&self) -> *const T;
+}
+
+impl<T> AsPtr<T> for Option<T> {
+    fn as_ptr(&self) -> *const T {
+        match self {
+            Some(v) => v,
+            None => std::ptr::null(),
+        }
+    }
+}
+
+trait AsMutPtr<T> {
+    fn as_mut_ptr(&mut self) -> *mut T;
+}
+
+impl<T> AsMutPtr<T> for Option<T> {
+    fn as_mut_ptr(&mut self) -> *mut T {
+        match self {
+            Some(v) => v,
+            None => std::ptr::null_mut(),
+        }
+    }
+}
+
+fn main() -> Result<(), String> {
+    // should we run only tests that shadow supports
+    let run_only_passing_tests = std::env::args().any(|x| x == "--shadow-passing");
+    // should we summarize the results rather than exit on a failed test
+    let summarize = std::env::args().any(|x| x == "--summarize");
+
+    let tests = if run_only_passing_tests {
+        get_passing_tests()
+    } else {
+        get_all_tests()
+    };
+
+    run_tests(tests.iter(), summarize)?;
+
+    println!("Success.");
+    Ok(())
+}
+
+fn get_passing_tests() -> std::collections::BTreeMap<String, TestFn> {
+    #[rustfmt::skip]
```

Comment:
So I went with this formatting for the tests since I found it much more readable than Rust's default (by default it will use 4 lines per test rather than our 2), but I don't think we should use this format for all lists/tuples. Maybe a macro would be better here, but I haven't yet figured out the macro syntax.